File: VB/WindowsApplication1/SkinnedStandardButton.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Drawing
Imports System.Windows.Forms
Imports DevExpress.XtraEditors.Drawing
Imports DevExpress.LookAndFeel
Imports DevExpress.XtraEditors.Controls
Imports DevExpress.Utils.Drawing
Imports DevExpress.Utils
Imports System.ComponentModel

Namespace WindowsApplication1
	<System.ComponentModel.DesignerCategory("")> _
	Public Class SkinnedStandardButton
		Inherits Button

		Private _Button As New EditorButton(ButtonPredefines.Glyph)
		Private _Painter As SkinEditorButtonPainter
		Public Sub New()
			_Painter = CreatePainter()
		End Sub

		Private _UseSkins As Boolean = True
		<DesignerSerializationVisibility(DesignerSerializationVisibility.Visible)> _
		Public Property UseSkins() As Boolean
			Get
				Return _UseSkins
			End Get
			Set(ByVal value As Boolean)
				_UseSkins = value
				Refresh()
			End Set
		End Property


		Protected Overridable Function CreatePainter() As SkinEditorButtonPainter
			Return New SkinEditorButtonPainter(UserLookAndFeel.Default)
		End Function

		Private Function GetObjectState() As ObjectState
			Dim location As Point = PointToScreen(New Point(0, 0))
			Dim mouseInButton As Boolean = New Rectangle(location, New Size(Width, Height)).Contains(Control.MousePosition)
			If (Not mouseInButton) OrElse DesignMode Then
				Return ObjectState.Normal
			End If
			If Control.MouseButtons = MouseButtons.Left Then
				Return ObjectState.Pressed
			Else
				Return ObjectState.Hot
			End If
		End Function
		Protected Overrides Sub OnPaint(ByVal pevent As PaintEventArgs)
			If UseSkins Then
				Dim args As New EditorButtonObjectInfoArgs(New GraphicsCache(pevent.Graphics), _Button, AppearanceObject.EmptyAppearance)
				args.Bounds = New Rectangle(0, 0, Width, Height)
				args.BuiltIn = False
				args.FillBackground = False
				args.State = GetObjectState()
				args.Button.Caption = Text
				_Painter.DrawObject(args)
			Else
				MyBase.OnPaint(pevent)
			End If
		End Sub


	End Class
End Namespace

File: CS/WindowsApplication1/SkinnedStandardButton.cs
using System;
using System.Drawing;
using System.Windows.Forms;
using DevExpress.XtraEditors.Drawing;
using DevExpress.LookAndFeel;
using DevExpress.XtraEditors.Controls;
using DevExpress.Utils.Drawing;
using DevExpress.Utils;
using System.ComponentModel;

namespace WindowsApplication1
{
    [System.ComponentModel.DesignerCategory("")]
    public class SkinnedStandardButton : Button
    {

        private EditorButton _Button = new EditorButton(ButtonPredefines.Glyph);
        private SkinEditorButtonPainter _Painter;
        public SkinnedStandardButton()
        {
            _Painter = CreatePainter();
        }

        private bool _UseSkins = true;
        [DesignerSerializationVisibility(DesignerSerializationVisibility.Visible)]
        public bool UseSkins
        {
            get { return _UseSkins; }
            set
            {
                _UseSkins = value;
                Refresh();
            }
        }
        

        protected virtual SkinEditorButtonPainter CreatePainter()
        {
            return new SkinEditorButtonPainter(UserLookAndFeel.Default);
        }

        private ObjectState GetObjectState()
        {
            Point location = PointToScreen(new Point(0, 0));
            bool mouseInButton = new Rectangle(location, new Size(Width, Height)).Contains(Control.MousePosition);
            if (!mouseInButton || DesignMode)
                return ObjectState.Normal;
            return Control.MouseButtons == MouseButtons.Left ? ObjectState.Pressed : ObjectState.Hot;
        }
        protected override void OnPaint(PaintEventArgs pevent)
        {
            if (UseSkins)
            {
                EditorButtonObjectInfoArgs args = new EditorButtonObjectInfoArgs(new GraphicsCache(pevent.Graphics), _Button, AppearanceObject.EmptyAppearance);
                args.Bounds = new Rectangle(0, 0, Width, Height);
                args.BuiltIn = false;
                args.FillBackground = false;
                args.State = GetObjectState();
                args.Button.Caption = Text;
                _Painter.DrawObject(args);
            }
            else
                base.OnPaint(pevent);
        }


    }
}
